check.go

```diff
@@ -89,22 +89,20 @@ func (c *CheckRunner) UpdateChecks(checks api.HealthChecks) {
 		}
 		if definition.HTTP != "" {
 			http := &consulchecks.CheckHTTP{
-				Notify:        c,
-				CheckID:       checkHash,
-				HTTP:          definition.HTTP,
-				Header:        definition.Header,
-				Method:        definition.Method,
-				TLSSkipVerify: definition.TLSSkipVerify,
-				Interval:      definition.IntervalDuration,
-				Timeout:       definition.TimeoutDuration,
-				Logger:        c.logger,
+				Notify:   c,
```

Comment:
`TLSSkipVerify` is now wrapped in a `TLSClientConfig`.

I think that rather than removing the field it should be wrapped in a simple TLS config the way we do here:
https://github.com/hashicorp/consul/blob/master/tlsutil/config.go#L592

That way we respect the `TLSSkipVerify` flag, which users still provide.

Reply:
I'll fix that. Thanks.

Reply:
Ok. All fixed. Thanks again.

Reply:
Great, thanks. I think L107 that was removed from this file also needs to be restored. Otherwise there are some updates to the definition that won't trigger an update in the check that is running.

However, instead of `TLSSkiPVerify` it would now reference `TLSClientConfig.InsecureSkipVerify`

Reply:
How half-assed of me. Let me fix that as well.

Reply:
Really glad you pointed that out... turns out that line had more than just the  TLSSkipVerify check on it. There were multiple things `&&` (anded) together on the same line, not just over multiple lines (like I read it). So I'm going to replace those as well and make it 1 per line to make that read cleaner.

Reply:
That should all be good now.